openmetadata-service/src/main/java/org/openmetadata/service/resources/query/QueryResource.java

```diff
@@ -143,7 +145,19 @@ public ResultList<Query> listQueries(
     if (!CommonUtil.nullOrEmpty(entityId)) {
       filter.addQueryParam("entityId", entityId.toString());
     }
-    return super.listInternal(uriInfo, securityContext, fieldsParam, filter, limitParam, before, after);
+    ResultList<Query> queries =
+        super.listInternal(uriInfo, securityContext, fieldsParam, filter, limitParam, before, after);
+    List<Query> maskedQueries =
```

Comment:
we shouldn't be filtering out queries after fetching them. This will cause issues with pagination. Above pagination call says 10 results returned and we filter out and UI shows as 10 results instead we display only 8 results